src/cfe.c

```diff
@@ -618,26 +638,25 @@ void et_from_soil(struct conceptual_reservoir *soil_res, struct evapotranspirati
     double Budyko_denominator;
     double Budyko;
     
-    if (et_struct->potential_et_m_per_timestep > 0){
+    if (et_struct->reduced_potential_et_m_per_timestep > 0){
         
         if (soil_res->storage_m >= soil_res->storage_threshold_primary_m){        
-            et_struct->actual_et_m_per_timestep = min(et_struct->potential_et_m_per_timestep, soil_res->storage_m);
-            soil_res->storage_m -= et_struct->actual_et_m_per_timestep;
-	    // Modified - no reason to set potential et to zero
-            et_struct->potential_et_m_per_timestep = et_struct->potential_et_m_per_timestep - et_struct->actual_et_m_per_timestep;
+            et_struct->actual_et_from_soil_m_per_timestep = min(et_struct->reduced_potential_et_m_per_timestep, soil_res->storage_m);
         }                 
         else if (soil_res->storage_m > soil_parms->wilting_point_m && soil_res->storage_m < soil_res->storage_threshold_primary_m){
         
             Budyko_numerator = soil_res->storage_m - soil_parms->wilting_point_m;
             Budyko_denominator = soil_res->storage_threshold_primary_m - soil_parms->wilting_point_m;
             Budyko = Budyko_numerator / Budyko_denominator;
-                           
-            et_struct->actual_et_m_per_timestep = Budyko * (et_struct->potential_et_m_per_timestep);
-                           
-            soil_res->storage_m -= et_struct->actual_et_m_per_timestep;
-            et_struct->potential_et_m_per_timestep = et_struct->potential_et_m_per_timestep - et_struct->actual_et_m_per_timestep;
+            // LKC: Include check to guarantee EAT is not larger than soil storage               
+            et_struct->actual_et_from_soil_m_per_timestep = min(Budyko * (et_struct->reduced_potential_et_m_per_timestep), soil_res->storage_m);
+                                                   
         }
+        

```

Comment:
Should there be another else if here?
else if (soil_res ->storage_m <= wilting_point){ et_struct->actual_et_from_soil_m_per_timestep = 0.0;}